src/cascadia/TerminalControl/SearchBoxControl.cpp

```diff
@@ -0,0 +1,144 @@
+﻿//
+// SearchBoxControl.cpp
+// Implementation of the SearchBoxControl class
+//
+
+#include "pch.h"
+#include "SearchBoxControl.h"
+#include "SearchBoxControl.g.cpp"
+
+using namespace winrt;
+using namespace winrt::Windows::UI::Xaml;
+
+namespace winrt::Microsoft::Terminal::TerminalControl::implementation
+{
+    SearchBoxControl::SearchBoxControl() :
+        _goForward(false),
+        _isCaseSensitive(false)
+    {
+        InitializeComponent();
+
+        _goForwardButton = this->FindName(L"SetGoForwardButton").try_as<Controls::Primitives::ToggleButton>();
+        _goBackwardButton = this->FindName(L"SetGoBackwardButton").try_as<Controls::Primitives::ToggleButton>();
+
+        // TO DO: Is there a general way to put all the focusable elements in the
+        // collection ? Maybe try DFS
+        auto closeButton = this->FindName(L"CloseButton").try_as<Controls::Button>();
+        auto textBox = this->FindName(L"TextBox").try_as<Controls::TextBox>();
+        auto checkBox = this->FindName(L"CaseSensitivityCheckBox").try_as<Controls::CheckBox>();
+        auto moveButton = this->FindName(L"MoveSearchBoxPositionButton").try_as<Controls::Primitives::ToggleButton>();
+
+        if (closeButton)
+            _focusableElements.insert(closeButton);
+        if (textBox)
+            _focusableElements.insert(textBox);
+        if (checkBox)
+            _focusableElements.insert(checkBox);
+        if (moveButton)
+            _focusableElements.insert(moveButton);
+        if (_goForwardButton)
+            _focusableElements.insert(_goForwardButton);
+        if (_goBackwardButton)
+            _focusableElements.insert(_goBackwardButton);
+    }
+
+    bool SearchBoxControl::GetGoForward()

```

Comment:
All of the functions here don't have a function description comment. Please add them in.

Reply:
I used to think there is no need to add descriptions because most methods are just handlers. But I added, thanks ! :)